engine/src/main/java/org/pentaho/di/trans/streaming/common/BaseStreamStepMeta.java

```diff
@@ -194,18 +210,20 @@ public String getFileName() {
       TransMeta transMeta = mappingMetaRetriever.get( this, repository, metaStore, space );
       if ( !StringUtil.isEmpty( getSubStep() ) ) {
         String realSubStepName = space.environmentSubstitute( getSubStep() );
-        rowMeta.addRowMeta( transMeta.getPrevStepFields( realSubStepName ) );
-        transMeta.getSteps().stream().filter( stepMeta -> stepMeta.getName().equals( realSubStepName ) )
-          .findFirst()
-          .ifPresent( stepMeta ->
-          {
-            try {
-              stepMeta.getStepMetaInterface()
-                .getFields( rowMeta, origin, info, nextStep, space, repository, metaStore );
-            } catch ( KettleStepException e ) {
-              throw new RuntimeException( e );
-            }
-          } );
+        if ( transMeta.getSteps().stream().anyMatch( stepMeta -> stepMeta.getName().equals( realSubStepName ) ) ) {
```

Comment:
Do we need an else here?  We still want an exception I think, it just needs to be more informative.  Perhaps just adding a log message to the "throw new RuntimeException" a few lines down?